### DISCUSSION TITLE
equal method adding an extra dimension to the variable

### DISCUSSION THREAD

**Discussion by Vulinux** [francof2a/fxpmath#96] [Q&A]
Hello,

I had a question about the equal method. Why does equal method add an extra dimension to the variable calling it? I provided a simple piece of code that demonstrates this. If I choose to set the value with the equal method, scalars become 1D arrays, 1D arrays become 2D arrays ([ ] -> [ [] ]). But when I use the set_val() method this doesn't happen. I am somehow not using these methods correctly but I don't know how. Any help would be greatly appreciated.

I am working in Windows 10 and fxpmath version is 0.4.9.

```python
import numpy as np
from fxpmath import Fxp

# Create a variable a.
a = Fxp(-0.635 * np.ones(3), dtype='fxp-s16/14')
print('a = ')
a.info()

# Create a variable b.
b = Fxp(0, like=a)
print('Original b = ')
b.info()

# Change the value of b to be equal to a in two different ways.
b.set_val(a)
# b.equal(a)
print('After changing value of b = ')
b.info()
print(b.shape)
```

**Comment by npcomplete2023** [comment]
Hey, I encountered the exact same problem where the result of the operation is encompassed in an array
To be specific, this phenomenon seems to occur when you pass to .equal()  a number in the Fxp format
(in contrast, if you pass floats, - it is parsed fine, with no resulting array)

Interestingly, when using set_val the operation seems Ok

Here is a specific code example  (in comments - the resulting variable after each line)
#################
A=Fxp(2.125)    #A fxp-s6/3(2.125)
B=Fxp(0.925)  #fxp-s53/52(0.925)
R1=Fxp(A*B,signed=True,n_frac=5,n_word=10)  #fxp-s10/5(1.9375)

R2=Fxp(None,signed=True,n_frac=5,n_word=10)  #fxp-s10/5([1.9375])
R2 = R2.equal(A*B) #fxp-s2/0([1.])

R3=Fxp(None,signed=True,n_frac=5,n_word=10)  #fxp-s10/5(1.9375)
R3.set_val(A*B)  #fxp-s10/5(1.9375)
################

Thank you
